### PR TITLE
plug some mem leaks

### PR DESCRIPTION
**Pull request by gnode1**

### CODE REVIEW

File: km/km_gdb_stub.c [hunk 1]
```diff
@@ -1408,6 +1413,10 @@ static void handle_qxfer_execfile_read(const char* packet, char* obuf)
       obuf[1 + bytes_to_deliver] = 0;
    }
    send_packet(obuf);
+   if (obuf[0] == 'l') {
+      free(gdb_execfile_name);
+      gdb_execfile_name = NULL;
```
Comment (paulpopelka) [inline km/km_gdb_stub.c:1418]: I think you need to set gdb_execfile_len to zero here too.

**Review by paulpopelka — APPROVED** [review]
Looks fine except for one comment that should be fixed.

File: km/km_gdb_stub.c (hunk 1 above)
Comment (gnode1) [reply km/km_gdb_stub.c:1418]: I guess. OTOH it seems to me the decision is based on `offset == 0`, in which case `gdb_execfile_len` is unconditionally recomputed. Am I missing something?

Comment (gnode1) [reply km/km_gdb_stub.c:1418]: The way I see it, if `gdb_execfile_name` is freed, three lines down `gdb_execfile_name` is assigned unconditionally. Of curse this depends on the client useing `offset` in a specific pattern.

I'll make the change as it certainly doesn't make anything worse. I'm just explaining the way I was thinking about it.

**Pull request merged**